test/README.md

```diff
@@ -0,0 +1,72 @@
+## Test Overview
```

Comment:
Wow, nice write-up .. appreciate this !  We could move to the Wiki or even a gh_pages branch if we end up changing frequently.